In [1]:
from astropy.io import ascii
data = ascii.read("ES.asc")
print(data)

  Date    Time   Open    High    Low    Close  Volume
-------- ----- ------- ------- ------- ------- ------
09/10/97 08:31  1089.0 1089.25  1089.0 1089.25      0
09/10/97 08:32  1089.0  1089.0 1088.75  1089.0      0
09/10/97 08:33 1088.75  1089.0  1088.5 1088.75      0
09/10/97 08:34  1089.0  1089.5  1089.0 1089.25      0
09/10/97 08:35 1089.25 1089.25  1088.5  1088.5      0
09/10/97 08:36 1088.25  1088.5 1087.75 1087.75      0
09/10/97 08:37  1088.0 1088.25  1088.0 1088.25      0
09/10/97 08:38 1088.25 1088.25  1087.5 1087.75      0
09/10/97 08:39  1087.5 1088.75  1087.5 1088.75      0
09/10/97 08:40  1088.5  1088.5  1087.5  1087.5      0
     ...   ...     ...     ...     ...     ...    ...
01/27/17 15:05  2289.5 2289.75 2289.25 2289.75   4163
01/27/17 15:06 2289.75 2289.75  2289.5  2289.5    529
01/27/17 15:07 2289.75 2289.75  2289.5  2289.5    854
01/27/17 15:08 2289.75 2289.75  2289.5  2289.5    986
01/27/17 15:09 2289.75 2289.75  2289.5 2289.75   1173
01/27/17 15:10  2289.5 2289.

In [128]:
data = data.to_pandas()

pandas.core.frame.DataFrame

In [20]:
import numpy as np
import pandas as pd

In [126]:
from __future__ import division
t1 = data.head(n=10)
t1["diff3"]=t1.groupby(['Date'])['Close'].diff(3)
for i in range(0 , len(t1)-3):
    t1.loc[i+3,'diff3'] = t1.iloc[i+3]['diff3']/t1.iloc[i]['Close']
    

/Users/lingxiangyu/anaconda/lib/python2.7/site-packages/ipykernel/__main__.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  app.launch_new_instance()


In [121]:
a = t1.iloc[2+3]['diff3']/t1.iloc[2]['Close']
a

-0.00091848450057405281

In [139]:
t1["diff3"]

0         NaN
1         NaN
2         NaN
3    0.000000
4   -0.000459
5   -0.000918
6   -0.000918
7   -0.000689
8    0.000919
9   -0.000689
Name: diff3, dtype: float64

In [137]:
a = t1["diff3"].std()
a

0.0006593184212526261

In [278]:
data1 = data.head(n=10000)
data1

Date   Time     Open     High      Low    Close  Volume
0     09/10/97  08:31  1089.00  1089.25  1089.00  1089.25       0
1     09/10/97  08:32  1089.00  1089.00  1088.75  1089.00       0
2     09/10/97  08:33  1088.75  1089.00  1088.50  1088.75       0
3     09/10/97  08:34  1089.00  1089.50  1089.00  1089.25       0
4     09/10/97  08:35  1089.25  1089.25  1088.50  1088.50       0
5     09/10/97  08:36  1088.25  1088.50  1087.75  1087.75       0
6     09/10/97  08:37  1088.00  1088.25  1088.00  1088.25       0
7     09/10/97  08:38  1088.25  1088.25  1087.50  1087.75       0
8     09/10/97  08:39  1087.50  1088.75  1087.50  1088.75       0
9     09/10/97  08:40  1088.50  1088.50  1087.50  1087.50       0
10    09/10/97  08:41  1087.50  1088.50  1087.50  1088.50       0
11    09/10/97  08:42  1088.75  1089.00  1088.25  1089.00       0
12    09/10/97  08:43  1088.75  1089.25  1088.50  1089.00       0
13    09/10/97  08:44  1089.00  1089.75  1089.00  1089.75       0
14    09/10/97  08:45  1089.75  1090.00  1089.50  1089.75       0
15    09/10/97  08:46  1089.75  1090.00  1089.75  1090.00       0
16    09/10/97  08:47  1090.25  1090.75  1090.00  1090.75       0
17    09/10/97  08:48  1091.25  1091.50  1090.50  1090.50       0
18    09/10/97  08:49  1090.25  1090.75  1090.00  1090.25       0
19    09/10/97  08:50  1090.25  1090.25  1090.00  1090.00       0
20    09/10/97  08:51  1090.00  1090.50  1090.00  1090.50       0
21    09/10/97  08:52  1090.75  1091.50  1090.75  1091.50       0
22    09/10/97  08:53  1091.25  1091.50  1090.75  1090.75       0
23    09/10/97  08:54  1091.00  1091.00  1090.50  1090.75       0
24    09/10/97  08:55  1091.00  1091.25  1090.50  1090.75       0
25    09/10/97  08:56  1090.75  1091.00  1090.00  1090.25       0
26    09/10/97  08:57  1090.25  1090.50  1090.25  1090.50       0
27    09/10/97  08:58  1090.50  1090.50  1090.50  1090.50       0
28    09/10/97  08:59  1090.75  1090.75  1090.50  1090.75       0
29    09/10/97  09:00  1090.75  1090.75  1090.25  1090.50       0
...        ...    ...      ...      ...      ...      ...     ...
9970  10/14/97  12:41  1118.75  1119.25  1118.75  1119.25       0
9971  10/14/97  12:42  1119.50  1119.50  1119.25  1119.25       0
9972  10/14/97  12:43  1119.50  1119.50  1119.25  1119.50       0
9973  10/14/97  12:44  1119.50  1119.75  1119.25  1119.50       0
9974  10/14/97  12:45  1119.50  1119.50  1119.00  1119.25       0
9975  10/14/97  12:46  1119.50  1119.50  1119.25  1119.25       0
9976  10/14/97  12:47  1119.25  1119.25  1119.00  1119.00       0
9977  10/14/97  12:48  1119.00  1119.25  1118.75  1118.75       0
9978  10/14/97  12:49  1118.75  1119.00  1118.50  1118.50       0
9979  10/14/97  12:50  1118.50  1118.75  1118.25  1118.50       0
9980  10/14/97  12:51  1118.50  1118.75  1118.50  1118.75       0
9981  10/14/97  12:52  1119.00  1119.00  1119.00  1119.00       0
9982  10/14/97  12:53  1119.25  1119.50  1119.25  1119.25       0
9983  10/14/97  12:54  1119.50  1119.75  1119.50  1119.75       0
9984  10/14/97  12:55  1120.00  1120.00  1120.00  1120.00       0
9985  10/14/97  12:56  1120.00  1120.00  1119.75  1119.75       0
9986  10/14/97  12:57  1119.75  1119.75  1119.25  1119.25       0
9987  10/14/97  12:58  1119.25  1119.50  1119.25  1119.50       0
9988  10/14/97  12:59  1119.50  1119.75  1119.00  1119.50       0
9989  10/14/97  13:00  1119.50  1119.75  1119.50  1119.50       0
9990  10/14/97  13:01  1119.50  1119.50  1119.25  1119.25       0
9991  10/14/97  13:02  1119.25  1119.25  1119.00  1119.25       0
9992  10/14/97  13:03  1119.00  1119.25  1119.00  1119.00       0
9993  10/14/97  13:04  1119.00  1119.00  1118.75  1119.00       0
9994  10/14/97  13:05  1119.00  1119.50  1118.75  1119.50       0
9995  10/14/97  13:06  1119.25  1119.50  1119.00  1119.25       0
9996  10/14/97  13:07  1119.00  1119.25  1118.75  1119.25       0
9997  10/14/97  13:08  1119.25  1120.25  1119.25  1120.25       0
9998  10/14/97  13:09  1120.25  1120.25  1119.50  

In [265]:
data1.iloc[405]['Close'] - data1.iloc[105]['Close']

-13.75

In [267]:
diff_temp1[405]

nan

In [285]:
data1.iloc[1000]['Close']

1070.0

diff_temp1 = data1.groupby(['Date'])['Close'].diff(300)
diff_ratio1 = pd.DataFrame(np.nan, index=np.arange(10000), columns=np.arange(1000))
#diff_ratio1[9999][998] = diff_temp1[9998]/data1.iloc[999]['Close']


diff_ratio = pd.DataFrame(np.nan, index=np.arange(10000), columns=np.arange(1000))
for i in range(1, 1000):
    diff_temp = data1.groupby(['Date'])['Close'].diff(i)
    for j in range(0,10000-i):
        diff_ratio[j+i][i] = diff_temp[j+i]/data1.iloc[j]['Close']

In [ ]:
diff_ratio = pd.DataFrame(np.nan, index=np.arange(10000), columns=np.arange(1000))
for i in range(0,999):
    for j in range(0,9999):
        if j > i:
           diff_ratio.loc[j][i] = (data1.iloc[j]['Close'] - data1.iloc[j-i-1]['Close'])/data.iloc[j-i-1]['Close']
#print diff_ratio        

In [181]:
t2 = pd.DataFrame(np.nan, index=np.arange(10), columns=np.arange(10))
t2

0   1   2   3   4   5   6   7   8   9
0 NaN NaN NaN NaN NaN NaN NaN NaN NaN NaN
1 NaN NaN NaN NaN NaN NaN NaN NaN NaN NaN
2 NaN NaN NaN NaN NaN NaN NaN NaN NaN NaN
3 NaN NaN NaN NaN NaN NaN NaN NaN NaN NaN
4 NaN NaN NaN NaN NaN NaN NaN NaN NaN NaN
5 NaN NaN NaN NaN NaN NaN NaN NaN NaN NaN
6 NaN NaN NaN NaN NaN NaN NaN NaN NaN NaN
7 NaN NaN NaN NaN NaN NaN NaN NaN NaN NaN
8 NaN NaN NaN NaN NaN NaN NaN NaN NaN NaN
9 NaN NaN NaN NaN NaN NaN NaN NaN NaN NaN

In [161]:
t2[1][1]

nan

In [3]:
print(colval)

 Close 
-------
1089.25
 1089.0
1088.75
1089.25
 1088.5
1087.75
1088.25
1087.75
1088.75
 1087.5
    ...
2289.75
 2289.5
 2289.5
 2289.5
2289.75
 2289.5
2289.25
 2289.5
2289.25
 2289.0
2288.75
Length = 1981244 rows


In [6]:
colval[0]

1089.25

In [7]:
type(colval)

astropy.table.column.Column

In [9]:
colval[1 to 2]

SyntaxError: invalid syntax (<ipython-input-9-893c1bb83669>, line 1)

In [10]:
colval[1,2]

IndexError: too many indices for array

In [18]:
import numpy as np
i=np.arange(2,10,1)

In [19]:
colval[i]

<Column name='Close' dtype='float64' length=8>
1088.75
1089.25
 1088.5
1087.75
1088.25
1087.75
1088.75
 1087.5

In [21]:
len(data)

1981244

In [29]:
t2 = data['Date' is '09/11/97']
print t2

  Date    Time  Open    High   Low    Close  Volume
-------- ----- ------ ------- ------ ------- ------
09/10/97 08:31 1089.0 1089.25 1089.0 1089.25      0


  Date    Time  Open    High   Low    Close  Volume
-------- ----- ------ ------- ------ ------- ------
09/10/97 08:31 1089.0 1089.25 1089.0 1089.25      0


In [66]:
t3 = t2.to_pandas()

In [87]:
t7.groupby(['Date'])['Close'].mean()

Date
09/10/97    1085.632099
09/11/97    1068.723457
Name: Close, dtype: float64

In [89]:
t7.groupby(['Date'])['Close'].var()

Date
09/10/97    10.16846
09/11/97    11.04524
Name: Close, dtype: float64